ts/src/pro/hashkey.ts

```diff
@@ -0,0 +1,852 @@
+        const orderbook = await this.wathPublic (market, topic, messageHash, params);
+        return orderbook.limit ();
+    }
+
+    handleOrderBook (client: Client, message) {
+        //
+        //     {
+        //         "symbol": "ETHUSDT",
+        //         "symbolName": "ETHUSDT",
+        //         "topic": "depth",
+        //         "params": { "realtimeInterval": "24h" },
+        //         "data": [
+        //             {
+        //                 "e": 301,
+        //                 "s": "ETHUSDT",
+        //                 "t": 1722873144371,
+        //                 "v": "84661262_18",
+        //                 "b": [
+        //                     [ "1650", "0.0864" ],
+        //                     ...
+        //                 ],
+        //                 "a": [
+        //                     ["4085", "0.0074" ],
+        //                     ...
+        //                 ],
+        //                 "o": 0
+        //             }
+        //         ],
+        //         "f": false,
+        //         "sendTime": 1722873144589,
+        //         "channelId": "2265aafffe68b588-00000001-0011510c-9e9ca710b1500854-551830bd",
+        //         "shared": false
+        //     }
+        //
+        const marketId = this.safeString (message, 'symbol');
+        const symbol = this.safeSymbol (marketId);
+        const messageHash = 'orderbook:' + symbol;
+        if (!(symbol in this.orderbooks)) {
+            this.orderbooks[symbol] = this.orderBook ({});
+        }
+        const orderbook = this.orderbooks[symbol];
+        const data = this.safeList (message, 'data', []);
+        const dataEntry = this.safeDict (data, 0);
+        const timestamp = this.safeInteger (dataEntry, 't');
+        const snapshot = this.parseOrderBook (dataEntry, symbol, timestamp, 'b', 'a');
+        orderbook.reset (snapshot);
+        orderbook['nonce'] = this.safeInteger (message, 'id');
+        this.orderbooks[symbol] = orderbook;
+        client.resolve (orderbook, messageHash);
+    }
+
+    async watchOrders (symbol: Str = undefined, since: Int = undefined, limit: Int = undefined, params = {}): Promise<Order[]> {
+        /**
+         * @method
+         * @name hashkey#watchOrders
+         * @description watches information on multiple orders made by the user
+         * @see https://hashkeyglobal-apidoc.readme.io/reference/websocket-api#private-stream
+         * @param {string} symbol unified market symbol of the market orders were made in
+         * @param {int} [since] the earliest time in ms to fetch orders for
+         * @param {int} [limit] the maximum number of order structures to retrieve
+         * @param {object} [params] extra parameters specific to the exchange API endpoint
+         * @returns {object[]} a list of [order structures]{@link https://docs.ccxt.com/#/?id=order-structure}
+         */
+        await this.loadMarkets ();
```

Comment:
will this work for both spot and swap orders?

Reply:
Let us check

Reply:
Yes it works for both types of markets